.travis.yml

```diff
@@ -16,34 +16,20 @@ services:
 stages:
   - name: Test and Lint
     if: type = pull_request
-  - name: Build
-    if: type = pull_request
   - name: Push
     if: type = push AND branch = master
-  - name: Codecov
-    if: type = pull_request
 
 jobs:
   include:
-    - stage: Build
-      name: Docker
-      script:
-        - TARGET=ship-it-api make build
-        - TARGET=ship-it-syncd make build
-    
-    - stage: Build
-      name: Operator
-      before_script:
-        - cd operator
-      script:
-        - make docker-build
-
     - stage: Test and Lint
       name: Go
-      install: GO111MODULE=off go get "github.com/golangci/golangci-lint/cmd/golangci-lint"
+      install: curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
       script:
         - golangci-lint --deadline=5m run -D errcheck -E goimports -E interfacer -E scopelint ./...
```

Comment:
I don't think golangci-lint needs to compile anything to run its checks, and probably doesn't share a lot of dependencies with our ship-it code. So I think we could split the stages like `Lint` and `Test and Build`, the advantage being that if the `Lint` stage fails travis won't even attempt to test and build.

Reply:
I think it would be even faster if we put everything (Lint, Test, Build) in 1 stage, in separate jobs. I checked our subscription and it looks like we are able to run 10 concurrent jobs. Let me know what you think.

Reply:
IMO I would stick to having the separate Lint and Test & Build stages. The Lint stage is the fastest stage, so there's quick PR feedback and since Travis runs the stages sequentially we can avoid testing/building code that doesn't pass lint.

A real downside is not seeing test failures if the lint stage fails, I think devs are/should be running their unit tests locally when they're changing code, but not everyone runs `golangci-lint` and with the exact same checks travis is running. So I'd rather have faster lint feedback even if it means sometime we don't let devs know about test failures as early as possible.